js/tests/enhancedDate.spec.js

```diff
@@ -0,0 +1,82 @@
+//suite description
+describe("Enhanced Date", function (){
+var testDate = enhancedDate.setDate(new Date(2015, 5, 30)); 
+    
+//  var enhancedDate;
+//  
+//  beforeEach(function () {
+//      enhancedDate = new enhancedDate();
+//  });
+
+//Get Day Name Test  
+  describe("Get Day Name", function(){
+    
+      it("should be a string and should return Tuesday", function(){
+          expect(enhancedDate.getDayName(testDate)).to.be.a("string").and.to.equal("Tuesday");
+      });
+  });
+
+
+//Get Month Name Test  
+  describe("Get Month Name", function(){
+    
+      it("should be a string and should return June", function(){
+          expect(enhancedDate.getMonthName(testDate)).to.be.a("string").and.to.equal("June");
+      });
+  });
+
+
+//Is Today Test  
+  describe("Is Today", function(){
+    
+      it("testDate should not be a string and should return false ", function(){
+          expect(enhancedDate.isToday(testDate)).to.not.be.an("string").and.to.not.be.true;
+      });
+    
+//      it("should not be a string ", function(){
+//          expect(enhancedDate.isToday(testDate)).to.not.be.an("string");
+//      });
+        
+  });
+
+
+//Is Future Test 
+  describe("Is Future", function(){ 
+
+      it("testDate should return true", function(){
+          expect(enhancedDate.isFuture(testDate)).to.be.true;
+      });
+    
+      it("should not be a string ", function(){
```

Comment:
To do it, you would need to call enhanced.setDate() with another value inside the test spec:

`it("should return true", function () { enhancedDate.setDate(/* some value that makes it true */); expect(enhancedDate.isFuture()).to.be.true; });`

then, for the next case:

`it("should return false", function () { enhancedDate.setDate(/* some value that makes it false */); expect(enhancedDate.isFuture()).to.be.false; });`